### DISCUSSION TITLE
Is ScoreReelController properly defined for logging ?

### DISCUSSION THREAD

**Discussion by mrechte** [missionpinball/missionpinball#120] [Q&A]
I added a logging section in config.yaml according to https://missionpinball.org/config/logging/ but whatever level I put, it does not change the log verbosity.

```
logging:
  file:
    score_reel_controller: full
```
Checking the code, I can see that ScoreReelController is not based on MpfController.
On top of that it is using self.log.debug instead of self.debug_log (as for example in SwitchController).

**Comment by toomanybrians** [comment]
Yeah this is another good find! Probably it’s not properly written.

I believe the score reel code is old. (I wrote it originally I think in 2014, I don’t know if it’s been updated since then.) I know the proper MpfController and logging was added later, so it’s possible it was never updated.

So like always, if you want to dive in and update it, feel free! But there’s a good chance that it’s old and doesn’t do things in the most up to date ways.

**Reply by mrechte** [reply]
Should not that file be located in core rather in devices directory ?

**Comment by mrechte** [comment]
Please, I need some clarification on LogMixin. Why do users of debug_log() have to test self._debug first ? Why is this not done directly in the debug_log implementation ? Same question for info_log().

I can see inconsistent usage depending on the module: some do test, some do not...

For instance I would code info_log() like:

```
    def info_log(self, msg: str, *args, context=None, error_no=None, **kwargs) -> None:
        """Log a message at the info level.

        Whether this message shows up in the console or log file is controlled
        by the settings used with configure_logging().
        """
        if not self.log:
            self._logging_not_configured()

        if self._info_to_console or self._debug_to_console:
            level = 22
        elif self._info_to_file or self._debug_to_file:
            level = 21
        else:
            return

        if not (self._info or self._debug) or not log.isEnabledFor(level):
            return

        self.log.log(level, self.format_log_line(msg, context, error_no), *args, **kwargs)

```

I also wonder how it is possible, for a same config_name, to have conflicting values for file and console. For instance if for switch_controller I set none for console and basic for file, I end up with INFO messages on console.
